main/app.py

```diff
@@ -0,0 +1,352 @@
+
+    def create_rooms(self, room):
+        """Allows user to enter a list of room names specifying
+                whether office or living spaces"""
+        room_type = raw_input(
+            "Enter room type: \n O: Office space \n L: Living space: \n")
+
+        while room_type != "O" and room_type != "L":
+            room_type = raw_input(
+                "Try again. Enter Room Type:\n O: Office space \n L: Living space: \n")
+        for room_name in room:
+            if room_type.upper() == 'O':
+                self.rooms[room_type.upper()].append(room_name)
+            else:
+                self.rooms['L'].append(room_name)
+            self.connect.execute(
+                "INSERT INTO Rooms (Name, type) VALUES (?, ?)", [room_name, room_type])
+            self.conn.commit()
+        print('New rooms succesfully created')
+        return room
+
+    def office_space_count(self, off_name):
+        '''Keep track of offices allocated should be less than 6'''
+        off_name = str(off_name)
+        office_space = self.connect.execute(
+            "SELECT COUNT(*) AS office_occupants FROM Persons  WHERE Persons.office_accommodation = ?", [off_name]).fetchall()
+        self.office_occupied = office_space[0][0]
+        return self.office_occupied
+
+    def living_space_count(self, liv_name):
+        '''Keep track of living space allocated should be less than 4'''
+        # print(liv_name)
+        living_space = self.connect.execute(
+            "SELECT COUNT(*) AS living_occupants FROM Persons WHERE Persons.living_accomodation = ?", [liv_name]).fetchall()
+        for liv_space in living_space:
+            self.living_occupied = liv_space[0]
+            print(self.living_occupied)
+        return self.living_occupied
+
+    def add_person(self, first_name, last_name, person_type, want_housing):
+        '''Method receives personnel data and calls other method to save and allocate rooms to personnel '''
+        name = first_name + " " + last_name
+        person_type = person_type
+        want_accommodation = want_housing
+        if person_type.upper() == 'STAFF':
+            self.people_data['Staff'].append(name)
+        else:
+            self.people_data['Fellow'].append({name: want_accommodation})
+
+        self.insert_db(name=name, person_type=person_type,
+                       want_accommodation=want_accommodation)
+
+        self.allocation_rule(name, person_type,
+                             want_accommodation)
+        return name +" " + person_type + " " + want_accommodation
+
+    def get_rooms(self, r_type):
+        return self.connect.execute(
+            "SELECT Name FROM Rooms where type = '" + r_type + "'").fetchall()
+
+    def allocation_rule(self, name, person_type, want_accommodation):
+        '''Randomly allocate everyone an office'''
+        office_rooms = self.get_rooms('O')
+        for office_room in office_rooms:
```

Comment:
**Code Quality: Critical**

This is extremely sub-optimal. You run this loop every time to get available offices. Can we have this information persistent somewhere and then remove an office/room as the case maybe when the room has been filled. That's a much better approach.